connections/etuovi/mapper.py

```diff
@@ -0,0 +1,242 @@
+) -> ExtraLink:
+    return ExtraLink(
+        link_url=link_url,
+        linktype_name=link_type,
+        link_urltitle=link_urltitle,
+    )
+
+
+def map_extra_links(elastic_apartment: Apartment) -> list:
+    extra_links = []
+    for field_name, link_type in EXTRA_LINK_MAPPING.items():
+        for link_url in handle_field(elastic_apartment[field_name]):
+            image = get_extra_link_mapping(link_url, link_type)
+            extra_links.append(image)
+    return extra_links
+
+
+def get_image_mapping(
+    image_type: RealtyImageType, image_seq: str, image_url: str
+) -> Image:
+    return Image(
+        image_desc="",
+        image_realtyimagetype=image_type,
+        image_seq=image_seq,
+        image_transfer_id=image_seq,
+        image_transfer_source=image_url,
+        image_url=image_url,
+    )
+
+
+def map_images(elastic_apartment: Apartment) -> list:
+    images = []
+    image_seq = 1
+    for field_name, image_type in IMAGE_MAPPING.items():
+        for image_url in handle_field(elastic_apartment[field_name]):
+            image = get_image_mapping(image_type, str(image_seq), image_url)
+            images.append(image)
+            image_seq += 1
+    return images
+
+
+def map_scontacts(elastic_apartment: Apartment) -> list:
+    return [
+        Scontact(
+            scontact_name=elastic_apartment["project_estate_agent"],
+            scontact_title="",
+            scontact_itempage_email=elastic_apartment["project_estate_agent_email"],
+            scontact_mobilephone="",
+            scontact_phone=elastic_apartment["project_estate_agent_phone"],
+            scontact_image_url="",
+        )
+    ]
+
+
+def get_elastic_value(
+    elastic_apartment: Apartment,
+    elastic_field: str,
+    elastic_value: Union[str, int, float, AttrList, datetime] = None,
+) -> Union[str, int, list, Decimal]:
+
+    if elastic_value is None:
+        elastic_value = getattr(elastic_apartment, elastic_field)
+
+    if isinstance(elastic_value, float):
+        return Decimal(elastic_value)
+    elif isinstance(elastic_value, AttrList):
+        return [
+            get_elastic_value(elastic_apartment, elastic_field, val)
+            for val in elastic_value
+        ]
+    elif isinstance(elastic_value, int) and elastic_field in PRICE_FIELDS:
+        # Prices are saved as cents in ElasticSearch. Convert to EUR.
+        return Decimal(elastic_value) / 100
+    else:
+        return elastic_value
+
+
+def map_item_fields(elastic_apartment: Apartment) -> dict:
+    item_dict = {}
+    for elastic_field, etuovi_field in ITEM_FIELDS.items():
+        elastic_value = get_elastic_value(elastic_apartment, elastic_field)
+        if not elastic_value:
+            continue
+        elif isinstance(etuovi_field, list):
+            for idx, field in enumerate(etuovi_field):
+                if len(elastic_value) >= idx + 1:
+                    item_dict[field] = elastic_value[idx]
```

Comment:
What's the logic behind checking the value length here?

Reply:
Yea, this is something I pondered for a while and I'm not sure if it is the best solution for this.

So the usecase for this is when there is a list field `showing_times` that needs to be split into multiple fields for the etuovi Item (`showingdate`, `showing_date2`).

So in the field mappings I put:
```"showing_times": ["showingdate", "showing_date2"]```

In the implementation I loop through the etuovi fields, aka ["showingdate", "showing_date2"], and map a value from the `showing_times` list for it:
```
            for idx, field in enumerate(etuovi_field):
                if len(elastic_value) >= idx + 1:
                    item_dict[field] = elastic_value[idx]
```

So we need the check that the length of the elastic value list is larger or equal than the mapped etuovi fields, in this case 2. So for example if there is only 1 value in the `showing_times` list, this checks that the index won't go out of range.

But now that I think about it, we can probably just replace this with:
```
item_dict.update(dict(zip(etuovi_field, elastic_value)))
```
:D

Reply:
Pushed the change. Good observation!